util/timer_test.cc

```diff
@@ -15,122 +15,56 @@ class TimerTest : public testing::Test {
 
  protected:
   std::unique_ptr<MockTimeEnv> mock_env_;
-
-#if defined(OS_MACOSX) && !defined(NDEBUG)
-  // On some platforms (MacOS) pthread_cond_timedwait does not appear
-  // to release the lock for other threads to operate if the deadline time
-  // is already passed. This is a problem for tests in general because
-  // TimedWait calls are a bad abstraction: the deadline parameter is
-  // usually computed from Env time, but is interpreted in real clock time.
-  // Since this test doesn't even pretend to use clock times, we have
-  // to mock TimedWait to ensure it yields.
-  void SetUp() override {
-    ROCKSDB_NAMESPACE::SyncPoint::GetInstance()->DisableProcessing();
-    ROCKSDB_NAMESPACE::SyncPoint::GetInstance()->ClearAllCallBacks();
-    ROCKSDB_NAMESPACE::SyncPoint::GetInstance()->SetCallBack(
-        "InstrumentedCondVar::TimedWaitInternal", [&](void* arg) {
-          uint64_t* time_us = reinterpret_cast<uint64_t*>(arg);
-          if (*time_us < mock_env_->RealNowMicros()) {
-            *time_us = mock_env_->RealNowMicros() + 1000;
-          }
-        });
-    ROCKSDB_NAMESPACE::SyncPoint::GetInstance()->EnableProcessing();
-  }
-#endif  // OS_MACOSX && !NDEBUG
-
   const uint64_t kSecond = 1000000;  // 1sec = 1000000us
+
+  void SetUp() override { mock_env_->InstallTimeWaitCallbackForMACOS(); }
 };
 
 TEST_F(TimerTest, SingleScheduleOnceTest) {
-  const int kIterations = 1;
-  uint64_t time_counter = 0;
-  mock_env_->set_current_time(0);
-
-  InstrumentedMutex mutex;
```

Comment:
This line seems suspicious to me (although you have removed it).
If we are going to use `MockTimeEnv`, shouldn't we specify the env when creating this `mutex`?